File: c/LeetCode_119_Pascals TriangleII.c
#include <stdio.h>
#include <stdlib.h>


int* getRow(int rowIndex, int* returnSize) {
    int *arr = (int*) malloc((rowIndex + 1) * sizeof(int));
    *returnSize = (rowIndex + 1);

    arr[0] = 1;
    int tmp = 1;
    for (int i = 1; i <= rowIndex; ++i) {
        int num = (tmp * (rowIndex - i + 1)) / i;
        arr[i] = num;
        tmp = num;
    }

    return arr;
}


int main() {
    int p;
    getRow(3, &p);

    return 0;
}

File: python/LeetCode_119_Pascals TriangleII.py
def getRow(rowIndex: int):
    arr = [1]
    for i in range(rowIndex + 1):
        for j in range(1, i + 1):
            if (j == i):
                arr.insert(0, 1)  
            else:
                arr[j - 1] = arr[j - 1] + arr[j]

    return arr

getRow(5)



# 0(N)
"""
def getRow(rowIndex: int):
    res = [1]
    prev = 1
    for k in range(1, rowIndex + 1):
        next_val = prev * (rowIndex - k + 1) // k
        res.append(next_val)
        prev = next_val
    return res
"""